common/src/main/java/org/dash/wallet/common/payments/parsers/Bech32.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2018 Coinomi Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dash.wallet.common.payments.parsers;
+
+import javax.annotation.Nullable;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import org.bitcoinj.core.AddressFormatException;
+
+import java.util.Arrays;
+import java.util.Locale;
+
+/**
+ * <p>Implementation of the Bech32 encoding.</p>
+ *
+ * <p>See <a href="https://github.com/bitcoin/bips/blob/master/bip-0350.mediawiki">BIP350</a> and
+ * <a href="https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki">BIP173</a> for details.</p>
+ */
+public class Bech32 {
+    /** The Bech32 character set for encoding. */
+    private static final String CHARSET = "qpzry9x8gf2tvdw0s3jn54khce6mua7l";
```

Comment:
this was taken from `bitcoinj` 0.16.x.  Newer bitcoin addresses use this format along with many other currencies.